raiden/network/proxies/channel_manager.py

```diff
@@ -44,38 +45,39 @@
 log = structlog.get_logger(__name__)  # pylint: disable=invalid-name
 
 
-class ChannelManager:
+class ChannelManager(ContractProxy):
```

Comment:
Why are you inheriting from `ContractProxy`, this expanding adding the proxy interface with implementation details that should not be used by client code.

Reply:
sorry I don't understand your comment. Proxies inherit `ContractProxy` that provides some compatibility methods, like en/de-coding of the arguments or local signing. It's logical that specific contract proxy is an extension of generic proxy, because they are, well, both providing contract functionality.

Reply:
why? just because of the name? we can rename the classes to something else than `proxy` if that is the case. Adding encoding/decoding and local signing to these objects is unnecessary.

Reply:
I still don't understand what's the problem here. If you feel like this PR needs changes, then please go ahead and suggest a concrete solution to the code I've pushed. Otherwise we're just wasting time obsessing over minute design details, which are probably correct in both cases.

Reply:
@pcppcp The question is what are you exactly using from the base class and why did you extend it?

Also `super().__init__(jsonrpc_client, contract)`

does not seem to match 
```
    def __init__(
            self,
            sender: Address,
            abi: Dict,
            contract_address: Address,
            call_function: Callable,
            transact_function: Callable,
            estimate_function: Optional[Callable] = None):
```

which is the constructor of the `ContractProxy` which is the super class.

Reply:
> @pcppcp The question is what are you exactly using from the base class and why did you extend it?

derived methods use `ContractProxy::transact()`. helper methods that do args sanitization are required to i.e. convert byte-encoded strings to web3-compatible address

>does not seem to match

constructor has changed:

```
class ContractProxy:
    def __init__(
        self,
        jsonrpc_client: JSONRPCClient,
        contract: Contract
    ):
``

Reply:
> constructor has changed:

Ah cool -- did not see that in the diff.

Reply:
> If you feel like this PR needs changes, then please go ahead and suggest a concrete solution to the code I've pushed.

Do *not* inherit form `ContractProxy`, please revert this change for all classes.